Generator/AssociationGenerator.php

```diff
@@ -0,0 +1,269 @@
+
+        foreach ($products as $i => $product) {
+            if (null !== $associationLimitNumber && $i >= $associationLimitNumber) {
+                break;
+            }
+
+            $association = [];
+            $association[$identifierCode] = $product->getIdentifier()->getData();
+
+            $associationTypeProduct = $this->getRandomAssociations('-products');
+            $randomProducts = $this->generateRandomProducts(
+                $product,
+                $productsToAssociate,
+                $associationsProductCount
+            );
+            if ('' !== $randomProducts) {
+                $association[$associationTypeProduct] = $randomProducts;
+            }
+
+            $associationTypeGroup = $this->getRandomAssociations('-groups');
+            $randomGroups = $this->generateRandomVariantGroups(
+                $associationsVariantGroupCount
+            );
+            if ('' !== $randomGroups) {
+                $association[$associationTypeGroup] = $randomGroups;
+            }
+
+            $this->bufferizeAssociation($association, $tmpFile);
+
+            $this->objectDetacher->detach($product);
+
+            $progress->advance();
+        }
+
+        $this->writeCsvFile($this->headers, $outputFile, $tmpFile, $delimiter);
+        unlink($tmpFile);
+
+        return $this;
+    }
+
+    /**
+     * Generates a list of random comma separated association types code
+     *
+     * @param $typeOfAssociation string
+     *
+     * @return array
+     */
+    protected function getRandomAssociations($typeOfAssociation)
+    {
+        $associationTypes = $this->faker->randomElements($this->associationTypeRepository->findAll());
+        $associationTypesCodes = array_map(function ($associationType) use ($typeOfAssociation) {
+            return $associationType->getCode() . $typeOfAssociation;
+        }, $associationTypes);
+
+        return join(',', $associationTypesCodes);
+    }
+
+    /**
+     * Find product skus different from the product passed as parameter
+     *
+     * @param ProductInterface $excludedProduct
+     * @param CursorInterface  $products
+     * @param int              $associationCount
+     *
+     * @return array
+     */
+    protected function generateRandomProducts($excludedProduct, CursorInterface $products, $associationCount)
+    {
+        $productIdentifiers = [];
+        $i = 0;
+
+        while ($i < $associationCount) {
+            if (!$products->valid()) {
+                $products->rewind();
+                $products->next();
+            }
+
+            $product = $products->current();
+
+            if ($excludedProduct->getIdentifier()->getData() !== $product->getIdentifier()->getData()) {
+                $productIdentifiers[] = $product->getIdentifier()->getData();
+                $i++;
+            }
+
+            $products->next();
+
+            $this->objectDetacher->detach($product);
+        }
+
+        return join(',', $productIdentifiers);
```

Comment:
please move this join, rename this method into generateRandomProductIdentifiers, and insert the join into line 113